common/charts/templates/secret.yaml

```diff
@@ -0,0 +1,19 @@
+{{/* 
+Secret for database
+- Uses annotation to prevent removal from zone=(test|prod)
+- Reuses existing passwords, if found
+*/}}
+---
+apiVersion: v1
+kind: Secret
+{{- include "metadata" . }}
+{{- if or (eq (.Values.zone | toString ) "test") (eq (.Values.zone | toString ) "prod") }}
+  annotations:
+    "helm.sh/resource-policy": "keep" # Retain secret for non-PRs (test|prod)
+{{- end }}
+stringData:
```

Comment:
I would suggest make everything as data, having two different for secrets is confusing, it will be b64encoded in either case.

Reply:
We're working with encoded data in one instance, but not the other.  It's all the same in the end, so if you think this is easier, then we can go with it.  Done!